contracts/interfaces/IVersionOracle.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier:MIT
+pragma solidity ^0.6.2;
+pragma experimental ABIEncoderV2;
+
+interface IVersionOracle {
+
+    //struct defining a single version entry
+    struct VersionInfo {
+        uint64 time;
+        bool canceled;
+        bytes32 version;
```

Comment:
Meanings of 'canceled' and 'version' fields are not clear to me at this point. Could you please elaborate? Thanks.